src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_grand_champions.cpp

```diff
@@ -80,43 +84,120 @@ enum Seats
     SEAT_ID_0                       = 0
 };
 
-/*
-struct Point
-{
-    float x, y, z;
-};
+Position const vehicleExitPos = { 759.680f, 665.237f, 412.392f };
 
-const Point MovementPoint[] =
+void EnterAggressiveMode(Creature* temp)
 {
-  {746.84f, 623.15f, 411.41f},
-  {747.96f, 620.29f, 411.09f},
-  {750.23f, 618.35f, 411.09f}
-};
-*/
-void AggroAllPlayers(Creature* temp)
+    temp->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_IMMUNE_TO_PC);
+    temp->SetReactState(REACT_AGGRESSIVE);
+}
+
+void DoCastPennant(Creature* me)
 {
-    Map::PlayerList const &PlList = temp->GetMap()->GetPlayers();
+    switch (me->GetEntry())
+    {
+        case NPC_MOKRA:
+            me->CastSpell(me, SPELL_FLAG_ORGRIMMAR);
+            break;
+        case NPC_ERESSEA:
+            me->CastSpell(me, SPELL_FLAG_SILVERMOON);
+            break;
+        case NPC_RUNOK:
+            me->CastSpell(me, SPELL_FLAG_THUNDER_BLUFF);
+            break;
+        case NPC_ZULTORE:
+            me->CastSpell(me, SPELL_FLAG_SENJIN);
+            break;
+        case NPC_VISCERI:
+            me->CastSpell(me, SPELL_FLAG_UNDERCITY);
+            break;
+        case NPC_JACOB:
+            me->CastSpell(me, SPELL_FLAG_STORMWIND);
+            break;
+        case NPC_AMBROSE:
+            me->CastSpell(me, SPELL_FLAG_GNOMEREGAN);
+            break;
+        case NPC_COLOSOS:
+            me->CastSpell(me, SPELL_FLAG_EXODAR);
+            break;
+        case NPC_JAELYNE:
+            me->CastSpell(me, SPELL_FLAG_DARNASSUS);
+            break;
+        case NPC_LANA:
+            me->CastSpell(me, SPELL_FLAG_IRONFORGE);
+            break;
+        default:
+            break;
+    }
+    if (me->GetVehicleBase())
+        me->CastSpell(me->GetVehicleBase(), SPELL_TRAMPLE_DUMMY_AURA, true);
+}
 
-    if (PlList.isEmpty())
-            return;
+Creature* FindMyMount(Creature* me)
+{
+    InstanceScript* instance = me->GetInstanceScript();
+    if (!instance)
+        return NULL;
 
-    for (Map::PlayerList::const_iterator i = PlList.begin(); i != PlList.end(); ++i)
+    if (me->GetGUID() == instance->GetGuidData(DATA_GRAND_CHAMPION_1))
     {
-        if (Player* player = i->GetSource())
-        {
-            if (player->IsGameMaster())
-                continue;
+        if (Creature* mount = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_GRAND_CHAMPION_VEHICLE_1)))
```

Comment:
the NULL check is not needed here, if it's NULL then https://github.com/TrinityCore/TrinityCore/pull/14938/files#diff-2e14db63ffb42614aa604bc50e478a31R160 will return NULL anyway.